src/UserInterface/Image/createComponentSelector.js

```diff
@@ -47,19 +63,22 @@ function createComponentSelector(store, imageUIGroup) {
     action(event => {
       event.preventDefault()
       event.stopPropagation()
-      store.imageUI.selectedComponentIndex = Number(event.target.value)
+      const selIdx = Number(event.target.dataset.componentIndex)
+      if (event.target.type === 'radio') {
+        store.imageUI.selectedComponentIndex = selIdx
+      } else if (event.target.type === 'checkbox') {
+        const visibility = event.target.checked ? 1.0 : 0.0
+        store.imageUI.componentVisibilities[selIdx] = visibility
+        updateSliceActors(store)
+        updateVolumeProps(store)
```

Comment:
We want these to calls to go into a reaction for `store.imageUI.componentVisibilities.slice()`. This way, they will always get called when the component visibilities are changes, which could happen in the user interface or from other sources, i.e. an API call that originates from Python activities.